lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -387,46 +418,67 @@ class StableHLOToTTIRReduceOpConversionPattern
     }
 
     op = op->getNextNode();
-    if (!isa<stablehlo::SelectOp>(op)) {
+    if (!op || !(isa<stablehlo::SelectOp>(op) || isa<stablehlo::MaxOp>(op))) {
       return false;
     }
 
     op = op->getNextNode();
-    if (!isa<stablehlo::CompareOp>(op)) {
+    if (!op || !isa<stablehlo::CompareOp>(op)) {
```

Comment:
Nit: you can modify this and all other similar checks in same way
```suggestion
    if (!isa_and_nonnull<stablehlo::CompareOp>(op)) {
```